WP-586: Improve Field().retrieve() Performance

timm3 opened the pull request:
Only fetch fields necessary for data retrieval instead of the entire record.

timm3 committed:
WP-586: Improve Field().retrieve() Performance

Only fetch fields necessary for data retrieval instead of the entire record.

john-pierce reviewed (changes requested):
Won't this give a `KeyError:` in the `authorize` case if updatefield is a bool or not a string?

I'm unsure what the behavior would have been prior because the resulting query would be `SELECT * ...` but it will definitely cause problems if this isn't accounted for now.

timm3 commented:
@john-pierce good catch. I'm looking into it.

timm3 committed:
WP-586 update: use old method if authorize needed

The `self.authorize` function is typically (always?) a custom function not provided by pydal or web2py, so one cannot anticipate its needs. Since this is the case, only retrieve the bare minimum of fields necessary IF custom authorization does not need to be done.
